Sources/ApexyURLSession/URLSessionClient.swift

```diff
@@ -142,3 +203,10 @@ enum URLSessionClientError: LocalizedError {
         }
     }
 }
+
+private struct ResponseObserverWrapper {
```

Comment:
I didn't understand the purpose of this structure.
Can you please explain ?

Reply:
If we don't use wrapper here
```swift
defer {
    let wrapper = ResponseObserverWrapper(request: request.request, data: response?.data, response: response?.response, error: error)
    completionQueue.async {
        self.responseObserver?(wrapper.request, wrapper.response as? HTTPURLResponse, wrapper.data, wrapper.error)
    }
}
```

we got error 

![Screenshot 2021-07-07 at 11 07 19](https://user-images.githubusercontent.com/4734983/124723301-93aea100-df13-11eb-844f-fc4b6b871544.png)

Reply:
@vani2 
You can do it with capture list in closure

```swift
        defer {
            completionQueue.async {[request, response, error] in
                self.responseObserver?(request, response?.response as? HTTPURLResponse, response?.data, error)
            }
        }
```